### PR TITLE
CSPL-1593: Add test where operator is reset while app install is in progress

### PR DESCRIPTION
**Pull request by tpereirasplunk**
S1, C3, M4 automation test covering the following cases:

 (install phase - operator reset):

- Create and apply a CR Spec with App Source containing 10 Apps 
- Wait for app installation to start 
- Kill the operator pod while app framework is in Install phase
- Wait for Operator Pod to come back up 
- Verify apps are installed correctly

(Download phase - operator reset):

- Create and apply a CR Spec with App Source containing 10 Apps 
- Wait for app installation to start 
- Kill the operator pod while app framework is in Install phase
- Wait for Operator Pod to come back up 
- Verify apps are installed correctly

INT test pass run: https://github.com/splunk/splunk-operator/actions/runs/2022325652.

### CODE REVIEW

File: test/m4/appframework/appframework_test.go [hunk 1]
```diff
@@ -1724,4 +1724,92 @@ var _ = Describe("m4appfw test", func() {
 			testenv.VerifyAppInstalled(deployment, testenvInstance, testenvInstance.GetName(), shcPodNames, appList, true, "enabled", false, true)
 		})
 	})
+
+	Context("Multisite Indexer Cluster with Search Head Cluster (m4) with App Framework", func() {
+		It("smoke, m4, appframeworkm4, appframework: can deploy a M4 SVA with App Framework enabled, install apps and upgrade them", func() {
```
Comment (pdhanoyasplunk) [inline test/m4/appframework/appframework_test.go:1729]: Do we need this in smoke ?

**Comment by tpereirasplunk** [comment]
> @tpereirasplunk
> 
> The common theme in the tests you have written appears to be to check if Verify App installation is in progress and the deleting the Operator pods. However the Story's acceptance criteria also expects to Delete the Operator Pod while the Download/Copy is happening (in addition to Install). How are these cases covered?

The function `VerifyAppInstallInProgress` called in the test cases actually checks if the operator is in between the pod download, copy or install phase (Basically before install complete). With the apps we currently have in the framework we are unable to drill down into an exact phase as it it only encountered for a sec. 

Once we have larger apps like ES app we might be able to drill into exact phase like copy or download

File: test/m4/appframework/appframework_test.go (hunk 1 above)
Comment (tpereirasplunk) [reply test/m4/appframework/appframework_test.go]: modified

**Comment by smohan-splunk** [comment]
> > @tpereirasplunk
> > The common theme in the tests you have written appears to be to check if Verify App installation is in progress and the deleting the Operator pods. However the Story's acceptance criteria also expects to Delete the Operator Pod while the Download/Copy is happening (in addition to Install). How are these cases covered?
> 
> The function `VerifyAppInstallInProgress` called in the test cases actually checks if the operator is in between the pod download, copy or install phase (Basically before install complete). With the apps we currently have in the framework we are unable to drill down into an exact phase as it it only encountered for a sec.
> 
> Once we have larger apps like ES app we might be able to drill into exact phase like copy or download

Thanks @tpereirasplunk . As discussed lets update the Story description to reflect this isn't a deep level check

**Review by smohan-splunk — APPROVED** [review]

**Review by jambrosiano — APPROVED** [review]

**Review by pdhanoyasplunk — APPROVED** [review]

File: test/testenv/verificationutils.go [hunk 2]
```diff
@@ -759,13 +759,13 @@ func VerifyAppListPhase(deployment *Deployment, testenvInstance *TestEnv, name s
 	}
 }
 
-// VerifyAppInstallInProgress verify given app installation is in progress, i.e when Status is between 101 and 303
-func VerifyAppInstallInProgress(deployment *Deployment, testenvInstance *TestEnv, name string, crKind string, appSourceName string, appList []string) {
+// VerifyAppState verify given app state is in between states passed as parameters, i.e when Status is between 101 and 303 we would pass enterpriseApi.AppPkgInstallComplete and enterpriseApi.AppPkgPodCopyComplete
+func VerifyAppState(deployment *Deployment, testenvInstance *TestEnv, name string, crKind string, appSourceName string, appList []string, appStateInitial enterpriseApi.AppPhaseStatusType, aapStateFinal enterpriseApi.AppPhaseStatusType) {
```
Comment (jambrosiano) [inline test/testenv/verificationutils.go:763]: nit: typo on aapStateFinal

**Pull request merged**